src/v/model/tests/ktp_test.cc

```diff
@@ -0,0 +1,274 @@
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
```

Comment:
The unit test for ktp, tests especially the hash and equivalence functions, including between types.